libfwupdplugin/fu-context.c

```diff
@@ -1020,12 +1020,17 @@ fu_context_detect_full_disk_encryption(FuContext *self)
 		GDBusProxy *proxy = g_ptr_array_index(devices, i);
 		g_autoptr(GVariant) id_type = g_dbus_proxy_get_cached_property(proxy, "IdType");
 		g_autoptr(GVariant) device = g_dbus_proxy_get_cached_property(proxy, "Device");
-		if (id_type == NULL || device == NULL)
-			continue;
-		if (g_strcmp0(g_variant_get_string(id_type, NULL), "BitLocker") == 0)
+		g_autoptr(GVariant) id_label = g_dbus_proxy_get_cached_property(proxy, "IdLabel");
+		if (id_type != NULL && device != NULL &&
+		    g_strcmp0(g_variant_get_string(id_type, NULL), "BitLocker") == 0)
 			priv->flags |= FU_CONTEXT_FLAG_FDE_BITLOCKER;
+
+		if (id_type != NULL && id_label != NULL &&
```

Comment:
`else if` here perhaps? i.e. it can't be _both_, right?

Reply:
It can. You could have two disks in the system.

Reply:
Fair point!